workflow/initiator_group_test.go

```diff
@@ -0,0 +1,144 @@
+package workflow
+
+import (
+	"github.com/hpe-storage/nimble-golang-sdk/pkg/client/v1/nimbleos"
+	"github.com/hpe-storage/nimble-golang-sdk/pkg/param"
+	"github.com/hpe-storage/nimble-golang-sdk/pkg/service"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"testing"
+)
+
+var initiatorGroupName = "InitiatorGroupTest"
+
+type IGWorkflowSuite struct {
+	suite.Suite
+	groupService        *service.NsGroupService
+	initiatorGrpService *service.InitiatorGroupService
+	volumeService       *service.VolumeService
+}
+
+func (suite *IGWorkflowSuite) SetupSuite() {
+	groupService, err := config()
+	assert.Nilf(suite.T(), err, "Unable to connect to group, err: %v", err)
+	suite.groupService = groupService
+	suite.initiatorGrpService = groupService.GetInitiatorGroupService()
+	suite.volumeService = groupService.GetVolumeService()
+	createDefaultVolume(suite.volumeService)
+}
+
+func (suite *IGWorkflowSuite) TearDownSuite() {
+	suite.deleteInitiatorGroup("TestIGIscsi")
+	suite.deleteInitiatorGroup(initiatorGroupName)
+	err := deleteDefaultVolume(suite.volumeService)
+	assert.Nilf(suite.T(), err, "Unable to delete default volume, err: %v", err)
+}
+
+func (suite *IGWorkflowSuite) deleteInitiatorGroup(igName string) {
+	ig, _ := suite.initiatorGrpService.GetInitiatorGroupByName(igName)
+	if ig != nil {
+		err := suite.initiatorGrpService.DeleteInitiatorGroup(*ig.ID)
+		assert.Nilf(suite.T(), err, "Unable to delete Initiator group, err: %v", igName)
+	}
+}
+
+func (suite *IGWorkflowSuite) TestCreateIGMissingParam() {
+	newIG := &nimbleos.InitiatorGroup{
+		Name: param.NewString(initiatorGroupName),
+	}
+	_, err := suite.initiatorGrpService.CreateInitiatorGroup(newIG)
+	assert.NotNil(suite.T(), err, "Initiator group creation should have failed with error missing param")
+}
+
+func (suite *IGWorkflowSuite) TestCreateIG() {
+	var desc *string = param.NewString("Workflow initiator group")
+	newIG := &nimbleos.InitiatorGroup{
+		Name:           param.NewString(initiatorGroupName),
+		Description:    desc,
+		AccessProtocol: nimbleos.NsAccessProtocolIscsi,
+	}
+	_, err := suite.initiatorGrpService.CreateInitiatorGroup(newIG)
+	assert.Nilf(suite.T(), err, "Unable to create initiator group, err: %v", err)
+	ig, _ := suite.initiatorGrpService.GetInitiatorGroupByName(initiatorGroupName)
+	assert.Equalf(suite.T(), *desc, *ig.Description, "Initiator group creation does not have expected description: %v", desc)
```

Comment:
We shouldn't need to include the IP address in the message.  The test framework will show us the expected and actual values.  This comment applies to all the assert.Equalf statements in this patch.  They can be changed to assert.Equal.